hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/ConcurrentDistributionSetInvalidationTest.java

```diff
@@ -38,35 +48,54 @@
+         * This gives this test more time to succeed
+         */
+        @Bean
+        @Primary
+        RolloutGroupRepository slowRolloutGroupRepository(final RolloutGroupRepository groupRepo) {
+            final RolloutGroupRepository slowGroupRepo = mock(RolloutGroupRepository.class, delegatesTo(groupRepo));
+
+            doAnswer(invocation -> {
+                final JpaRolloutGroup group = invocation.getArgument(0);
+                if (group.getId() == null) {
+                    return groupRepo.save(group);
+                }
+                TimeUnit.SECONDS.sleep(2);
+                return groupRepo.save(group);
+            }).when(slowGroupRepo).save(any());
+
+            return slowGroupRepo;
+        }
+    }
 
     @Test
     @Description("Verify that a large rollout causes a timeout when trying to invalidate a distribution set")
     public void verifyInvalidateDistributionSetWithLargeRolloutThrowsException() throws Exception {
         final DistributionSet distributionSet = testdataFactory.createDistributionSet();
-        testdataFactory.createTargets(10000, "verifyInvalidateDistributionSetWithLargeRolloutThrowsException");
-        final RolloutGroupConditions conditions = new RolloutGroupConditionBuilder().withDefaults()
-                .successCondition(RolloutGroupSuccessCondition.THRESHOLD, "50")
-                .errorCondition(RolloutGroupErrorCondition.THRESHOLD, "80")
-                .errorAction(RolloutGroupErrorAction.PAUSE, null).build();
-        final Rollout rollout = rolloutManagement.create(entityFactory.rollout().create()
-                .name("verifyInvalidateDistributionSetWithLargeRolloutThrowsException").description("desc")
-                .targetFilterQuery("name==*").set(distributionSet).actionType(ActionType.FORCED), 20, conditions);
+        final Rollout rollout = createRollout(distributionSet);
         final String tenant = tenantAware.getCurrentTenant();
 
         // run in new Thread so that the invalidation can be executed in
         // parallel
-        final Thread handleRolloutsThread = new Thread(() -> {
-            tenantAware.runAsTenant(tenant, () -> systemSecurityContext.runAsSystem(() -> {
-                rolloutManagement.handleRollouts();
-                return 0;
-            }));
-        });
-        handleRolloutsThread.start();
+        new Thread(() -> systemSecurityContext.runAsSystemAsTenant(() -> {
+            rolloutManagement.handleRollouts();
+            return 0;
+        }, tenant)).start();
+
         // wait until at least one RolloutGroup is created, as this means that
         // the thread has started and has acquired the lock
-        Awaitility.await().until(() -> tenantAware.runAsTenant(tenant, () -> systemSecurityContext
+        Awaitility.await().atMost(Duration.FIVE_SECONDS)
+                .pollInterval(Duration.ONE_HUNDRED_MILLISECONDS)
+                .until(() -> tenantAware.runAsTenant(tenant, () -> systemSecurityContext
```

Comment:
could we also use `systemSecurityContext.runAsSystemAsTenant` as above?

Reply:
I did not change the logic, I only added the `atMost()` and the `pollInterval()` to have a faster feedback